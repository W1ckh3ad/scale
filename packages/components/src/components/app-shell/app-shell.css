.sr-only {
  position: absolute;
  left: -10000px;
  overflow: hidden;
}

.content {
  background: var(--scl-color-background-standard, #fff);
  padding-left: var(--header-content-side-padding, 24px);
  padding-right: var(--header-content-side-padding, 24px);
}

.shell {
  --header-nav-height: 56px;
  --header-brand-height: 72px;
  --header-border-radius: 8px;
  --header-transition-speed: 0.3s;
  --header-brand-collapsed-height: 4px;
  --header-max-width: inherit;
}
@media (max-width: 1023px) {
  .shell .content {
    padding-top: calc(
      var(--header-brand-collapsed-height) + var(--header-nav-height)
    );
  }
}
@media (min-width: 1024px) {
  .shell.shell--sticky .content {
    padding-top: calc(
      var(--header-brand-collapsed-height) + var(--header-nav-height)
    );
  }
  .content {
    transition: padding-top var(--header-transition-speed) ease-in-out;
    padding-top: calc(var(--header-brand-height) + var(--header-nav-height));
  }
}
@media (min-width: 1552px) {
  .shell .content {
    margin: 0 auto;
    max-width: var(--header-max-width);
  }
}
