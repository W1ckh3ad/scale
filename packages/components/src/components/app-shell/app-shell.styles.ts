import { JssStyle } from 'jss';

const contentPadding = {
  '& $content': {
    paddingTop:
      'calc(var(--header-brand-collapsed-height) + var(--header-nav-height))',
  },
};

export const styles: JssStyle = {
  content: {
    background: '#fff',
  },
  shell: {
    '--header-brand-height': '72px',
    '--header-brand-collapsed-height': '4px',
    '--header-nav-height': '56px',
    '--header-transition-speed': '0.3s',
    '--header-border-radius': '8px',
    '@media (max-width: 1023px)': {
      ...contentPadding,
    },
  },
  '@media (min-width: 1024px)': {
    shell: {
      '&.sticky': {
        ...contentPadding,
      },
    },
    content: {
      paddingTop: 'calc(var(--header-brand-height) + var(--header-nav-height))',
      transition: 'padding-top var(--header-transition-speed) ease-in-out',
    },
  },
};
