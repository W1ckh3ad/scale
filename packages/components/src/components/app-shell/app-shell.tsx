import { Component, h, Prop, Host, State, Listen } from '@stencil/core';
import classNames from 'classnames';
import { CssClassMap } from '../../utils/utils';

const levelEnhancer = (data, i = 0) => {
  let parsedData;
  try {
    parsedData = JSON.parse(data);
  } catch (error) {
    parsedData = data;
  }
  return parsedData.reduce((previous, current) => {
    // console.log('reducing');
    const reducer = (prev, curr, level) => [...prev, { ...curr, level }];
    if (current.children) {
      const enhancedCurr = {
        ...current,
        children: levelEnhancer(current.children, i + 1),
      };
      return reducer(previous, enhancedCurr, i + 1);
    }
    return reducer(previous, current, i + 1);
  }, []);
};

@Component({
  tag: 'scale-app-shell',
  styleUrl: 'app-shell.css',
  shadow: true,
})
export class Shell {
  @Prop() brandTitle?: string = '';
  @Prop() mainNavigation?: any = [];
  @Prop() iconNavigation?: any = [];
  @Prop() sectorNavigation?: any = [];
  @Prop() addonNavigation?: any = [];
  @Prop() customClass?: string = '';
  @Prop() activeRouteId?: string = '';
  @Prop() activeSectorId?: string = '';
  @State() scrolled: boolean = false;

  @Listen('scroll', { target: 'window' })
  onScroll() {
    this.scrolled = window.pageYOffset > 2;
  }

  componentWillLoad() {}
  componentWillUpdate() {}
  componentDidUnload() {}

  render() {
    return (
      <Host>
        <div class={this.getCssClassMap()}>
          <scale-app-header
            scrolled={this.scrolled}
            brandTitle={this.brandTitle}
            mainNavigation={levelEnhancer(this.mainNavigation)}
            iconNavigation={levelEnhancer(this.iconNavigation)}
            sectorNavigation={levelEnhancer(this.sectorNavigation)}
            addonNavigation={levelEnhancer(this.addonNavigation)}
            activeRouteId={this.activeRouteId}
            activeSectorId={this.activeSectorId}
          ></scale-app-header>
          <main class="content">
            <slot></slot>
          </main>
        </div>
      </Host>
    );
  }

  getCssClassMap(): CssClassMap {
    return classNames(
      'shell',
      this.customClass && this.customClass,
      this.scrolled && 'sticky'
    );
  }
}
