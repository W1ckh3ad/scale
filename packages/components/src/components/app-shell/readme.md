# scale-shell



<!-- Auto Generated Below -->


## Properties

| Property           | Attribute           | Description | Type     | Default |
| ------------------ | ------------------- | ----------- | -------- | ------- |
| `activeRouteId`    | `active-route-id`   |             | `string` | `''`    |
| `activeSectorId`   | `active-sector-id`  |             | `string` | `''`    |
| `addonNavigation`  | `addon-navigation`  |             | `any`    | `[]`    |
| `brandTitle`       | `brand-title`       |             | `string` | `''`    |
| `claimLang`        | `claim-lang`        |             | `string` | `'de'`  |
| `customClass`      | `custom-class`      |             | `string` | `''`    |
| `iconNavigation`   | `icon-navigation`   |             | `any`    | `[]`    |
| `mainNavigation`   | `main-navigation`   |             | `any`    | `[]`    |
| `portalName`       | `portal-name`       |             | `string` | `''`    |
| `sectorNavigation` | `sector-navigation` |             | `any`    | `[]`    |


## Dependencies

### Depends on

- [scale-app-header](../app-header)

### Graph
```mermaid
graph TD;
  scale-app-shell --> scale-app-header
  scale-app-header --> app-mega-menu
  scale-app-header --> scale-icon
  scale-app-header --> app-logo
  scale-app-header --> app-navigation-sector-mobile
  scale-app-header --> app-navigation-main-mobile
  app-navigation-main-mobile --> scale-icon
  style scale-app-shell fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
