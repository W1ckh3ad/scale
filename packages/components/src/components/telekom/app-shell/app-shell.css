:host {
  --background: var(--scl-color-background-standard, #fff);
  --spacing-x: var(--scl-spacing-24);
  --radius: var(--scl-radius-8);

  --header-nav-height: 56px;
  --header-brand-height: 72px;
  --header-border-radius: var(--radius);
  --header-transition-speed: 0.3s;
  --header-brand-collapsed-height: 4px;
  --header-max-width: inherit;
}
.sr-only {
  position: absolute;
  left: -10000px;
  overflow: hidden;
}

.content {
  background: var(--background);
  padding-left: var(--spacing-x);
  padding-right: var(--spacing-x);
}

@media (max-width: 1023px) {
  .shell .content {
    padding-top: calc(
      var(--header-brand-collapsed-height) + var(--header-nav-height)
    );
  }
}
@media (min-width: 1024px) {
  .shell.shell--sticky .content {
    padding-top: calc(
      var(--header-brand-collapsed-height) + var(--header-nav-height)
    );
  }
  .content {
    transition: padding-top var(--header-transition-speed) ease-in-out;
    padding-top: calc(var(--header-brand-height) + var(--header-nav-height));
  }
}
@media (min-width: 1552px) {
  .shell .content {
    margin: 0 auto;
    max-width: var(--header-max-width);
  }
}
