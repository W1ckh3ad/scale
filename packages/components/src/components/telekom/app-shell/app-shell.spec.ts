import { newSpecPage } from '@stencil/core/testing';
import { Shell } from './app-shell';

describe('Shell', () => {
  let page;
  beforeEach(async () => {
    page = await newSpecPage({
      components: [Shell],
      html: `<scale-app-shell></scale-app-shell>`,
    });
  });
  it('should match snapshot', async () => {
    expect(page.root).toMatchSnapshot();
  });
});
