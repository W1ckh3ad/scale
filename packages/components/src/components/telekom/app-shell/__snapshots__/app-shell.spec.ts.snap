// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shell should match snapshot 1`] = `
<scale-app-shell>
  <mock:shadow-root>
    <div class="shell" part="base">
      <scale-app-header activerouteid="" activesectorid="" claimlang="de" portalname=""></scale-app-header>
      <main class="content">
        <slot></slot>
      </main>
    </div>
  </mock:shadow-root>
</scale-app-shell>
`;
